app/models/subnet.rb

```diff
@@ -274,6 +283,16 @@ def subnet_for(ip)
       ip = IPAddr.new(ip)
       Subnet.all.detect {|s| s.family == ip.family && s.contains?(ip)}
     end
+
+    # allows to create a specific subnet class based on the network_type.
+    # network_type is more user friendly than the class names
+    def new_network_type(args)
+      network_type = args.delete(:network_type) || N_('IPv4')
```

Comment:
no need for `N_(..)` here, it's already extracted in SUBNET_TYPES and will just return the same string